server/auth/registration.go

```diff
@@ -0,0 +1,42 @@
+package auth
+
+import (
+	"errors"
+
+	"github.com/8tomat8/SSU-Golang-252-Chat/loger"
+	"github.com/jinzhu/gorm"
+)
+
+// UserRegistry is a structure for Messager
+type UserRegistry struct {
+	UserName string `json:"user_name"`
+	Password string `json:"password"`
+	NickName string `json:"nick_name"`
+}
+
+type User struct {
+	UserName string `gorm:"not null;unique_index"`
+	Password string
+	NickName string
+}
+
+type UserGorm struct {
+	*gorm.DB
+}
+
+func (ug *UserGorm) Create(user *User) error {
+	newUser := ug.NewRecord(user)
```

Comment:
> [NewRecord check if value's primary key is blank](https://godoc.org/github.com/jinzhu/gorm#DB.NewRecord)

So it would not show you if user already exist.